test/parallel/test-assert-if-error.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+require('../common');
+const assert = require('assert').strict;
+/* eslint-disable no-restricted-properties */
+
+// Test that assert.ifError has the correct stack trace of both stacks.
+
+let err;
+// Create some random error frames.
+(function a() {
+  (function b() {
+    (function c() {
+      err = new Error('test error');
+    })();
+  })();
+})();
+
+const msg = err.message;
+const stack = err.stack;
+
+(function x() {
+  (function y() {
+    (function z() {
+      let threw = false;
+      try {
+        assert.ifError(err);
+      } catch (e) {
+        assert.equal(e.message, 'ifError got unwanted exception: test error');
+        assert.equal(err.message, msg);
+        assert.equal(e.actual, err);
+        assert.equal(e.actual.stack, stack);
+        assert.equal(e.expected, null);
+        assert.equal(e.operator, 'ifError');
+        assert(!e.stack.includes('at ifError'), `Got:\n\n  ${e.stack}\n\nAt:`);
+        assert(e.stack.includes('at z'), `Got:\n\n  ${e.stack}\n\nAt:`);
+        assert(e.stack.includes('at c'), `Got:\n\n  ${e.stack}\n\nAt:`);
+        assert(e.stack.indexOf('at z') < e.stack.indexOf('at c'));
+        threw = true;
+      }
+      assert(threw);
+    })();
+  })();
+})();
```

Comment:
I would prefer a test showing a full stack trace, rather than this check. It would mean future edits are easier to understand and fix.

Reply:
I added a separate test to check for the stack trace and minimized this test to the necessary parts.